src/ripple/app/tx/impl/CreateTicket.cpp

```diff
@@ -117,21 +117,17 @@ CreateTicket::doApply()
     for (std::uint32_t i = 0; i < ticketCount; ++i)
     {
         std::uint32_t const curTicketSeq = firstTicketSeq + i;
-
-        SLE::pointer sleTicket = std::make_shared<SLE>(
-            ltTICKET, getTicketIndex(account_, curTicketSeq));
+        Keylet const ticketKeylet = keylet::ticket(account_, curTicketSeq);
+        SLE::pointer sleTicket = std::make_shared<SLE>(ticketKeylet);
 
         sleTicket->setAccountID(sfAccount, account_);
         sleTicket->setFieldU32(sfTicketSequence, curTicketSeq);
         view().insert(sleTicket);
 
-        auto const page = dirAdd(
-            view(),
+        auto const page = view().dirInsert(
             keylet::ownerDir(account_),
-            sleTicket->key(),
-            false,
-            describeOwnerDir(account_),
-            viewJ);
+            ticketKeylet,
+            describeOwnerDir(account_));
```

Comment:
Nit: it may be worth replacing sleTicket->key() in the log call below with the new local ticketKeylet.key.